src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -738,6 +738,59 @@ def makeTransferRucio(self, wflow, dataIn, subLevel, blocks, dataSize, nodes, no
+        """
+        Send alert to Prometheus, wrap function in a try-except clause
+        """
+        try:
+            # alert to expiry in an hour from now
+            self.alertManagerApi.sendAlert(alertName, severity, summary, description,
+                                           service, endSecs)
+        except Exception as ex:
+            self.logger.exception("Failed to send alert to %s. Error: %s", self.alertManagerUrl, str(ex))
+
+    def alertPUMisconfig(self, workflowName):
+        """
+        Send alert to Prometheus with PU misconfiguration error
+        """
+        alertName = "{}: PU misconfiguration error. Workflow: {}".format(self.alertServiceName,
+                                                                         workflowName)
+        alertSeverity = "high"
+        alertSummary = "[MSTransferor] Workflow cannot proceed due to some PU misconfiguration."
+        alertDescription = "Workflow: {} could not proceed due to some PU misconfiguration,".format(workflowName)
+        alertDescription += "so it will be skipped."
+        self.sendAlert(alertName, alertSeverity, alertSummary, alertDescription,
+                       self.alertServiceName)
+        self.logger.critical(alertDescription)
+
+    def alertUnknownTransferError(self, workflowName):
+        """
+        Send alert to Prometheus with unknown transfer error
+        """
+        alertName = "{}: Transfer request error. Workflow: {}".format(self.alertServiceName,
+                                                                         workflowName)
+        alertSeverity = "high"
+        alertSummary = "[MSTransferor] Unknown exception while making Transfer Request."
+        alertDescription = "Unknown exception while making Transfer request for workflow: {}".format(workflowName)
+        self.sendAlert(alertName, alertSeverity, alertSummary, alertDescription,
+                       self.alertServiceName)
+        msg = "Unknown exception while making Transfer Request for %s " % wflow.getName()
+        self.logger.exception(alertDescription)
+
+    def alertTransferCouchDBError(self, workflowName):
+        """
+        Send alert to Prometheus with CouchDB transfer error
+        """
+        alertName = "{}: Transfer document error posting to CouchDB. Workflow: {}".format(self.alertServiceName,
+                                                                         workflowName)
+        alertSeverity = "high"
+        alertSummary = "[MSTransferor] Transfer document could not be created in CouchDB."
+        alertDescription = "Workflow: {}, failed request  due to error posting to CouchDB".format(workflowName)
+        self.sendAlert(alertName, alertSeverity, alertSummary, alertDescription,
+                       self.alertServiceName)
+        self.logger.warning(alertDescription)
+
+
     def notifyLargeData(self, aboveWarningThreshold, transferId, wflowName, dataSize, dataIn):
```

Comment:
Kenyi, since you will have to make some changes to this PR, can I ask you to provide an enhancement for this function as well? :-D
Could you please rename it to `alertLargeInputData`? Just so it's more consistent with the other alerts.